connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java

```diff
@@ -598,6 +601,120 @@ public void testAddRemoveTask() throws Exception {
+
+        EasyMock.expect(plugins.currentThreadLoader()).andReturn(delegatingLoader).times(1);
+        PowerMock.expectNew(WorkerSourceTask.class, EasyMock.eq(TASK_ID),
+            EasyMock.eq(task),
+            anyObject(TaskStatus.Listener.class),
+            EasyMock.eq(TargetState.STARTED),
+            anyObject(JsonConverter.class),
+            anyObject(JsonConverter.class),
+            anyObject(JsonConverter.class),
+            EasyMock.eq(new TransformationChain<>(Collections.emptyList(), NOOP_OPERATOR)),
+            anyObject(KafkaProducer.class),
+            anyObject(OffsetStorageReader.class),
+            anyObject(OffsetStorageWriter.class),
+            EasyMock.eq(config),
+            anyObject(ClusterConfigState.class),
+            anyObject(ConnectMetrics.class),
+            anyObject(ClassLoader.class),
+            anyObject(Time.class),
+            anyObject(RetryWithToleranceOperator.class)).andReturn(workerTask);
+        Map<String, String> origProps = new HashMap<>();
+        origProps.put(TaskConfig.TASK_CLASS_CONFIG, TestSourceTask.class.getName());
+
+        TaskConfig taskConfig = new TaskConfig(origProps);
+        // We should expect this call, but the pluginLoader being swapped in is only mocked.
+        // EasyMock.expect(pluginLoader.loadClass(TestSourceTask.class.getName()))
+        //        .andReturn((Class) TestSourceTask.class);
+        EasyMock.expect(plugins.newTask(TestSourceTask.class)).andReturn(task);
+        EasyMock.expect(task.version()).andReturn("1.0");
+
+        workerTask.initialize(taskConfig);
+        EasyMock.expectLastCall();
+
+        // Expect that the worker will create converters and will find them using the current classloader ...
+        assertNotNull(taskKeyConverter);
+        assertNotNull(taskValueConverter);
+        assertNotNull(taskHeaderConverter);
+        expectTaskKeyConverters(ClassLoaderUsage.CURRENT_CLASSLOADER, taskKeyConverter);
+        expectTaskValueConverters(ClassLoaderUsage.CURRENT_CLASSLOADER, taskValueConverter);
+        expectTaskHeaderConverter(ClassLoaderUsage.CURRENT_CLASSLOADER, taskHeaderConverter);
+
+        EasyMock.expect(executorService.submit(workerTask)).andReturn(null);
+
+        EasyMock.expect(plugins.delegatingLoader()).andReturn(delegatingLoader);
+        EasyMock.expect(delegatingLoader.connectorLoader(WorkerTestConnector.class.getName()))
+            .andReturn(pluginLoader);
+        EasyMock.expect(Plugins.compareAndSwapLoaders(pluginLoader)).andReturn(delegatingLoader)
+            .times(1);
+
+        EasyMock.expect(Plugins.compareAndSwapLoaders(delegatingLoader)).andReturn(pluginLoader)
+            .times(1);
+        plugins.connectorClass(WorkerTestConnector.class.getName());
+        EasyMock.expectLastCall().andReturn(WorkerTestConnector.class);
+
+        EasyMock.expect(workerTask.awaitStop(EasyMock.anyLong())).andStubReturn(true);
+        EasyMock.expectLastCall();
+
```

Comment:
Took me a little bit to understand when these were actually called: from within the `assertStatusMetrics(...)` method calls. Maybe a simple comment would make that more obvious:
```suggestion

        // Each time we check the task metrics, the worker will call the herder
```